S4TF_Tutorial_3.ipynb

```diff
@@ -0,0 +1,1769 @@
+        "colab": {}
+      },
+      "source": [
+        "public extension String {\n",
+        "    @discardableResult\n",
+        "    func shell(_ args: String...) -> String\n",
+        "    {\n",
+        "        let (task,pipe) = (Process(),Pipe())\n",
+        "        task.executableURL = URL(fileURLWithPath: self)\n",
+        "        (task.arguments,task.standardOutput) = (args,pipe)\n",
+        "        do    { try task.run() }\n",
+        "        catch { print(\"Unexpected error: \\(error).\") }\n",
+        "\n",
+        "        let data = pipe.fileHandleForReading.readDataToEndOfFile()\n",
+        "        return String(data: data, encoding: String.Encoding.utf8) ?? \"\"\n",
+        "    }\n",
+        "}"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "wwJGmDrQ0EoB"
+      },
+      "source": [
+        "This guide uses the [Fashion MNIST](https://github.com/zalandoresearch/fashion-mnist) dataset, which contains 70,000 grayscale images in 10 categories. The images show individual articles of clothing at low resolution (28 $\\times$ 28 pixels), as seen here:\n",
+        "\n",
+        "<table>\n",
+        "  <tr><td>\n",
+        "    <img src=\"https://tensorflow.org/images/fashion-mnist-sprite.png\"\n",
+        "         alt=\"Fashion MNIST sprite\"  width=\"600\">\n",
+        "  </td></tr>\n",
+        "  <tr><td align=\"center\">\n",
+        "    <b>Figure 1.</b> <a href=\"https://github.com/zalandoresearch/fashion-mnist\">Fashion-MNIST samples</a> (by Zalando, MIT License).<br/>&nbsp;\n",
+        "  </td></tr>\n",
+        "</table>\n",
+        "\n",
+        "Fashion MNIST is intended as a drop-in replacement for the classic [MNIST](http://yann.lecun.com/exdb/mnist/) dataset—often used as the \"Hello, World\" of machine learning programs for computer vision. The MNIST dataset contains images of handwritten digits (0, 1, 2, etc) in an identical format to the articles of clothing we'll use here.\n",
+        "\n",
+        "This guide uses Fashion MNIST for variety, and because it's a slightly more challenging problem than regular MNIST. Both datasets are relatively small and are used to verify that an algorithm works as expected. They're good starting points to test and debug code. \n",
+        "\n",
+        "We will use 60,000 images to train the network and 10,000 images to evaluate how accurately the network learned to classify images."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "Q_XrcNGNbejX",
+        "colab_type": "code",
+        "outputId": "1b76b5ec-992b-4af8-f912-2db4b9c0bdd7",
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 154
+        }
+      },
+      "source": [
+        "public func downloadFile(_ url: String, dest: String? = nil, force: Bool = false) {\n",
```

Comment:
- Avoid abbreviations in argument labels.
- `url:` should be an argument label.
```suggestion
        "public func downloadFile(url: String, destination: String? = nil, force: Bool = false) {\n",
```